### PR TITLE
ci: replace circleci with github actions

### PR DESCRIPTION
**Pull request by tac0turtle**
Signed-off-by: Marko Baricevic <marbar3778@yahoo.com>

* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGES.md

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      #     zip -0 ccov.zip `find . \( -name "*tendermint*.gc*" \) -print`
+      #     grcov ccov.zip -s . -t lcov --llvm --branch --ignore-not-existing --ignore "/*" -o lcov.info
+      #     bash <(curl -s https://codecov.io/bash) -f lcov.info
+      #     rm -f lcov.info
+      #     rm -f ccov.zip
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          # file: /tmp/grcov*
```
Comment (tarcieri) [inline .github/workflows/rust.yml]: Try this:

```suggestion
          file: ${{ steps.coverage.outputs.report }}
```

**Comment by liamsi** [comment]
The uploading to codecov is failing because the script needs a an ENV var with the token. I don't know how to add these in GA.

**Comment by tarcieri** [comment]
From the tab bar for the repo: Settings > Secrets

**Comment by liamsi** [comment]
@ebuchman to the rescue ...

**Comment by tac0turtle** [comment]
will close this PR and reopen it against the tendermint-rs repo

**Pull request closed without merging**